pallets/dapp-staking-v3/src/lib.rs

```diff
@@ -516,6 +516,12 @@ pub mod pallet {
     #[pallet::genesis_build]
     impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
         fn build(&self) {
+            if self == &Default::default() {
+                // Default config is an empty config, no build is needed
+                // Remove this is default genesis changes
+                return;
+            }
```

Comment:
@Dinonard a build-in test is added which tests default config https://github.com/paritytech/polkadot-sdk/blob/02e50adf7ba6cc65a9ef5c332b3e2974c8d23f48/substrate/frame/support/procedural/src/construct_runtime/expand/config.rs#L102
So we either have implement a default or do the above approach. Otherwise it will assert because current default derive isn't valid

Reply:
I see!

Let's make an issue for this and implement it later.

Reply:
If I had to suggest something, I'd suggest this:
* evenly split portions based on the number of tiers (should be at least 1 tier)
* thresholds can be static ones, hardcoded to whatever
* number of slots can also be hardcoded to whatever

The important thing is to adjust number of vector entries to the number of tiers, that's it.

Reply:
I think I did exactly like this, it was not pretty but worked